pyrasterframes/src/main/python/pyrasterframes/rf_ipython.py

```diff
@@ -152,9 +170,16 @@ def _safe_geom_to_html(g):
         else:
             return g.__repr__()
 
+    def _safe_bytearray_to_html(b):
+        if isinstance(b, bytearray):
+            return binary_to_html(b)
+        else:
+            return b.__repr__()
+
     # dict keyed by column with custom rendering function
     formatter = {c: _safe_tile_to_html for c in tile_cols}
     formatter.update({c: _safe_geom_to_html for c in geom_cols})
+    formatter.update({c: _safe_bytearray_to_html for c in bytearray_cols})
```

Comment:
So is this the magic; a place where you can register column formatters based on data type?

Reply:
You register a formatter for each column (`Series`)

As you see we look at the first row to see what the type is. If the type is interesting we put it in these dicts for the appropriate type.